File: src/main/java/gay/sylv/wij/impl/util/Initializable.java
/**
 * World In a Jar
 * Copyright (C) 2024  VulpixelMC
 * <p>
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * <p>
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 * <p>
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package gay.sylv.wij.impl.util;

import java.lang.reflect.InvocationTargetException;

public interface Initializable {
	default void initialize() {
		/* no-op */
	}
	
	static <T extends Initializable> void initialize(Class<T> clazz) {
		try {
			clazz.getConstructor().newInstance().initialize();
		} catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
			throw new RuntimeException(e);
		}
	}
}

File: src/main/kotlin/dev/cursedmc/wij/api/Initializable.kt
/**
 * Copyright (c) 2022 CursedMC. All rights reserved.
 *
 * World In A Jar is common software: you can redistribute it and/or modify it under the terms of the Commons Protection License as published by the Revolutionary Technical Committee.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the Commons Protection License for more details.
 */
@file:Suppress("unused")

package dev.cursedmc.wij.api

/**
 * @see gay.sylv.wij.api.Initializable
 */
@Deprecated(message = "Moved to gay.sylv.wij.api", level = DeprecationLevel.ERROR)
interface Initializable {
	fun initialize() {}
}
